src/libsyntax/parse/parser.rs

```diff
@@ -637,7 +637,26 @@ impl<'a> Parser<'a> {
                 let mut err = self.fatal(&format!("expected `{}`, found `{}`",
                                                   token_str,
                                                   this_token_str));
-                err.span_label(self.span, format!("expected `{}`", token_str));
+
+                let sp = if self.token == token::Token::Eof {
+                    // EOF, don't want to point at the following char, but rather the last token
+                    self.prev_span
```

Comment:
Is there a test reflecting this change?

Reply:
It seems like I didn't add one. It was a drive-by fix for a diagnostic I saw while fixing the rest to unify the output with the other, more general `unexpected token` error. I'll add one after fixing the wording for `Sized`.

Reply:
Added test.